state-chain/chains/src/arb.rs

```diff
@@ -66,16 +67,22 @@ impl Chain for Arbitrum {
 	fn input_asset_amount_using_reference_gas_asset_price(
 		input_asset: Self::ChainAsset,
 		required_gas: Self::ChainAmount,
+		oracle_price: Option<Price>,
```

Comment:
Could we try to move the querying of the price *into* this function? As it is now, to verify correctness, one has to look at all the call sites of this function, to make sure that the correct price for the correct asset is passed in, and there are no typing guardrails to prevent one from using the wrong asset for example.